client.go

```diff
@@ -239,21 +237,21 @@ func (c *client) handlePacket(remoteAddr net.Addr, packet []byte) error {
 		// otherwise this might be an attacker trying to inject a PUBLIC_RESET to kill the connection
 		if cr.Network() != remoteAddr.Network() || cr.String() != remoteAddr.String() || hdr.ConnectionID != c.connectionID {
```

Comment:
Remark: it's trivial to just spoof the sender address, so the only really relevant thing here is the connection id, right?